stl/inc/format

```diff
@@ -169,6 +169,14 @@ concept _CharT_or_bool = same_as<_Ty, _CharT> || same_as<_Ty, bool>;
 template <class _CharT>
 concept _Format_supported_charT = _Is_any_of_v<_CharT, char, wchar_t>;
 
+template <class _Ty, class _Context>
+concept _Has_formatter = requires(_Ty& _Val, _Context& _Ctx) {
+    _STD declval<typename _Context::template formatter_type<remove_cvref_t<_Ty>>>().format(_Val, _Ctx);
+};

```

Comment:
Why these requires' parameters have reference type? Isn't it redundant?

> parameter-list — a comma-separated list of parameters like in a function declaration, except that default arguments are not allowed and it cannot end with an ellipsis (other than one signifying a pack expansion). **These parameters have no storage, linkage or lifetime, and are only used to assist in specifying requirements.** These parameters are in scope until the closing } of the requirement-seq

https://en.cppreference.com/w/cpp/language/constraints

Reply:
They inhibit array-to-pointer and function-to-pointer decay, and they ensure that e.g. `decltype((_Val))` is an lvalue reference type whether `_Ty` is a reference or not. In this particular use, we only care about the decay.